src/legacy/core_plugins/kibana/public/dashboard/legacy_app.js

```diff
@@ -33,6 +33,7 @@ import { DashboardListing, EMPTY_FILTER } from './listing/dashboard_listing';
 import { addHelpMenuToAppChrome } from './help_menu/help_menu_util';
 import { registerTimefilterWithGlobalStateFactory } from '../../../../ui/public/timefilter/setup_router';
 import { syncOnMount } from './global_state_sync';
+import { savedObjectLoaderDashboard } from './saved_dashboard/saved_dashboards';
```

Comment:
This import isn't allowed because it uses services directly without going through `plugin.ts` (the linting PR will start flagging this).

Suggestion to do this cleanly:
* Create the saved_dashboard class and service in the start phase of the dashboard plugin and
* expose it through the contract.
* Then have a little wrapper in legacy world registering the class from the shimmed contract into angular for the management view.

This would put the class creation where it belongs, doesn't cross legacy/shimmed boundaries, wouldn't block moving the dashboard app into NP and is still compatible with the angular setup.

Basically it would mean moving this part into the `plugin.ts`:
```ts
const savedObjectsClient = npStart.core.savedObjects.client;
const services = {
  savedObjectsClient,
  indexPatterns: npStart.plugins.data.indexPatterns,
  chrome: npStart.core.chrome,
  overlays: npStart.core.overlays,
};

const SavedDashboard = createSavedDashboardClass(services);
export const savedObjectLoaderDashboard = new SavedObjectLoader(
  SavedDashboard,
  savedObjectsClient,
  npStart.core.chrome
);
```

And keeping just this part in legacy `saved_dasboard`:
```ts
// This is the only thing that gets injected into controllers
module.service('savedDashboards', () => dashboardStart.savedObjectLoaderDashboard);
```

Reply:
Thx, yes, the great listing improvement will complain about this changes. So I've solved it the "Discover way"™" for now, services created twice, and will clean this by removing `module.service('savedDashboards'...` in a follow up PR